### PR TITLE
Integrate fhirstore

### PR DESCRIPTION
**Pull request by simonvadee**
## Description

This PR replace fhirbase by MongoDB, using our wrapper `fhirstore`.

### Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation if any
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective, if unittests are enabled
- [x] New and existing unit tests pass locally with my changes, if any

### CODE REVIEW

File: config.yml [hunk 1]
```diff
@@ -1,28 +1,21 @@
...
-    fhirbase:
-        lib: 'psycopg2'
-        args: []
-        kwargs:
-            host: "localhost"
-            port: 5433
-            database: "fhirbase"
-            user: "postgres"
-            password: "postgres"
-    oracle:
```
Comment (LaRiffle) [inline config.yml]: I think we want to keep oracle: it's supposed to be an alternative driver to connect to oracle databases when it's not in postgres

File: config_docker.yml [hunk 2]
```diff
@@ -9,19 +9,12 @@ sql:
...
-  fhirbase:
-    lib: 'psycopg2'
-    args: []
-    kwargs:
-      host: 'fhir-pipe-fhirbase'
-      port: 5432
-      database: 'fhirbase'
-      user: 'postgres'
-      password: 'postgres'
-  oracle:
```
Comment (LaRiffle) [inline config_docker.yml]: Same here I think we want to keep oracle

File: docker-compose.yml [hunk 3]
```diff
@@ -31,17 +30,17 @@ services:
...
-      - '3000:3000'
-      - '5433:5432'
+      - 27017:27017
+    volumes:
+      - fhirstore-db:/data/db
 
 volumes:
   mimic-db:
     name: fhir-pipe-mimic-db
+  fhirstore-db:
```
Comment (LaRiffle) [inline docker-compose.yml]: Should this be there?

File: fhirpipe/load/sql.py [hunk 4]
```diff
@@ -2,38 +2,16 @@
...
-
-
 def get_connection(connection_type: str = None):
     """
-    Return a sql connexion depending on the configuration provided in config.yml
-    (see root of the project)
-    Note: should be used in a context environment (with get_connection(c) as ...)
+    Return a sql connexion depending on the configuration provided in
+    config.yml (see root of the project)
+    It should be used in a context environment(with get_connection(c) as ...)
```
Comment (LaRiffle) [inline fhirpipe/load/sql.py]: missing space before `(`

File: fhirpipe/parse/fhir.py [hunk 5]
```diff
@@ -170,27 +182,33 @@ def bind_reference(fhir_object, fhir_spec):
...
-        identifier = fhir_object['identifier']['value']
+    if fhir_object and fhir_object["identifier"]["value"]:
+        identifier = fhir_object["identifier"]["value"]
 
         # Collect all the resource_types which could be referenced
         try:
-            resource_types = re.search(
-                '\((.*)\)', fhir_spec["type"]).group(1).split('|')
+            resource_types = (
+                re.search("\((.*)\)", fhir_spec["type"]).group(1).split("|")
```
Comment (LaRiffle) [inline fhirpipe/parse/fhir.py]: Maybe we need to put the regexp out of the function to compile it a single time?
```
pattern = re.compile(r'([A-Z]+)([0-9]+)')
...
re.search(pattern, fhir_spec["type"]).group(1).split("|")
```

File: config.yml (hunk 1 above)
Comment (simonvadee) [reply config.yml]: ok, I removed it since it was not used in the code but it makes sense to keep it indeed.

Comment (simonvadee) [reply config.yml]: Done.

File: config_docker.yml (hunk 2 above)
Comment (simonvadee) [reply config_docker.yml]: Done.

File: fhirpipe/load/sql.py (hunk 4 above)
Comment (simonvadee) [reply fhirpipe/load/sql.py]: :+1:

File: fhirpipe/parse/fhir.py (hunk 5 above)
Comment (simonvadee) [reply fhirpipe/parse/fhir.py]: yes, not sure it's very expensive but it cannot hurt. Done.

**Pull request merged**